### PR TITLE
Added MaxRowCountConstraint

### PR DESCRIPTION
**Pull request by anev**
Added MaxRowCountConstraint annotation to control SqlQuery result max size. 
issue #216

### CODE REVIEW

**Comment by qualidafial** [comment]
Hi Andrei! Thanks for contributing.

What's your use case for this pull request? Have you considered using a limit clause in the SQL itself?

    @SqlQuery("select id, name from something where name = :name limit :maxRows")
    Something findByNameNew(@Bind("name") String name, @Bind("maxRowCount") long maxRows);

Or perhaps you're looking to verify that an expected number of rows match a query? In that case I suggest a `select count(*)` query instead:

    @SqlQuery("select count(*) from something where name = :name")
    long countByName(@Bind("name") String name);

File: src/main/java/org/skife/jdbi/v2/sqlobject/ResultReturnThing.java [hunk 1]
```diff
@@ -102,11 +104,32 @@ else if (return_type.isInstanceOf(Iterator.class)) {
...
         }
 
         @Override
         protected Object result(ResultBearing q, HandleDing baton)
         {
+            if (maxRowCount == 0) {
+                return returnFirst(q);
+            }
+
+            if (q.list(maxRowCount + 1).size() > maxRowCount) {
```
Comment (qualidafial) [inline src/main/java/org/skife/jdbi/v2/sqlobject/ResultReturnThing.java]: This is very inefficient. You are executing the query and pulling down all data in all rows just to verify the row count. Better to use a `select count(*)` query instead and avoid all the network and disk i/o.

Comment (stevenschlansker) [inline src/main/java/org/skife/jdbi/v2/sqlobject/ResultReturnThing.java]: It also means the query will be executed twice, I think?  That could be quite dangerous if it is e.g. an `UPDATE ... RETURNING *` style of statement.

**Comment by stevenschlansker** [comment]
Hi @anev we didn't hear back from you so I'm going to close this PR.  If you're still interested in pursuing this let us know the problem you are trying to solve -- I believe it is strictly better to use SQL `LIMIT` so the database knows what's up.  We certainly can reopen the issue if you have more to add.

**Pull request closed without merging**